packages/styled/types/base.d.ts

```diff
@@ -99,23 +99,23 @@ export interface CreateStyledComponent<
  */
 export interface CreateStyled {
   <
-    C extends React.ComponentType<React.ComponentProps<C>>,
+    C extends React.ComponentType<any>,
     ForwardedProps extends keyof React.ComponentProps<
       C
     > = keyof React.ComponentProps<C>
   >(
     component: C,
-    options: FilteringStyledOptions<PropsOf<C>, ForwardedProps>
+    options: FilteringStyledOptions<React.ComponentProps<C>, ForwardedProps>
```

Comment:
this is not an equivalent, but actually I think the new version is more correct - we don't need to resolve defaultProps here (I don't care as much about resolving propTypes)